src/builtins/dec.zig

```diff
@@ -266,11 +275,20 @@ pub const RocDec = extern struct {
         return .{ .value = RocDec{ .num = answer[0] }, .has_overflowed = answer[1] == 1 };
     }
 
-    pub fn add(self: RocDec, other: RocDec) RocDec {
+    pub fn add(
+        self: RocDec,
+        other: RocDec,
+        roc_ops: *RocOps,
+    ) RocDec {
         const answer = RocDec.addWithOverflow(self, other);
 
         if (answer.has_overflowed) {
-            roc_panic("Decimal addition overflowed!", 0);
+            const utf8_bytes = "Decimal addition overflowed!";
+            const roc_crashed_args = RocCrashed{
+                .utf8_bytes = @constCast(utf8_bytes.ptr),
+                .len = utf8_bytes.len,
+            };
+            roc_ops.roc_crashed(&roc_crashed_args, roc_ops.env);
```

Comment:
Up to you, and maybe better in a follow up, but might be nice to add helper functions to the `RocOps` struct. Then you can do something like:
```suggestion
            roc_ops.crash("Decimal addition overflowed!");
```

Have that function deal with all the abi stuff instead of repeat it every time you call the function.

Reply:
Nice idea... done in commit 5a80bb82f4a1629a61d51fbd35def17d7cf5ec4d